torch/csrc/jit/ir.h

```diff
@@ -60,6 +61,65 @@ static inline bool operator==(const Use & a, const Use & b) {
 // Graph holds a list of parameters.
 struct Param;
 
+// Scope is a node of a trie that represents the tree of nested scopes.
+// Individual scopes are pushed and popped from Graph, which hold a
+// pointer to the current scope. Each Node in Graph holds a pointer
+// to the scope that was current when the node was created.
+// The trie never needs to shrink, it only grows until it is disposed
+// of when Graph is deallocated. Hence, pointers to scopes held by nodes
+// will always be valid.
```

Comment:
It's worth noting that the ownership model as specified in the destructor implies that you cannot share common subtries between tries.

Reply:
Exactly, I couldn't think of a reason why different Graphs might share portions of the trie. I'm writing a note.